app/controllers/publishers/site_banners_controller.rb

```diff
@@ -5,14 +5,104 @@ class Publishers::SiteBannersController < ApplicationController
 
   MAX_IMAGE_SIZE = 10_000_000
 
-  def new
-    @site_banner = current_publisher.site_banner || SiteBanner.new
+  #Fetch Banner by channel_id, if it does not exist create a template banner.
+  def fetch
+    site_banner = current_publisher.site_banners.where(channel_id: params[:channel_id]).first
+    if site_banner.present?
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = data[:backgroundUrl]
+      data[:logoImage] = data[:logoUrl]
+      render(json: data.to_json)
+    else
+      site_banner = SiteBanner.new
+      site_banner.update(
+        publisher_id: current_publisher.id,
+        channel_id: params[:channel_id],
+        title: 'Brave Rewards',
+        description:
+'Thanks for stopping by. We joined Brave\'s vision of protecting your privacy because we believe that fans like you would support us in our effort to keep the web a clean and safe place to be.
+
+Your tip is much appreciated and it encourages us to continue to improve our content.',
+        donation_amounts: [1, 5, 10],
+        default_donation: 5,
+        social_links: {'youtube': '', 'twitter': '', 'twitch': ''}
+      )
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = nil
+      data[:logoImage] = nil
+      render(json: data.to_json)
+    end
   end
 
-  def create
-    site_banner = current_publisher.site_banner || SiteBanner.new
+  def channels
+
+    # Handle edge case where user hasn't created any channels yet
+    if current_publisher.channels.length == 0
+      data = {}
+      data[:channels] = [{'id' => '00000000-0000-0000-0000-000000000000', 'name' => ' ', 'type' => 'website'}]
+      data[:channel_mode] = false
+      data[:channel_index] = 0
+      render(json: data.to_json) and return
+    end
+
+    if current_publisher.site_banners.where(channel_id: '00000000-0000-0000-0000-000000000000').exists?
+      current_publisher.site_banners.where(channel_id: '00000000-0000-0000-0000-000000000000').first.delete
+    end
+
+    channels = []
+
+    current_publisher.channels.each do |c|
+
+    case c.details_type
+
+      when "SiteChannelDetails"
```

Comment:
So we have the same code in the `channels_controller` is there a reason why this auth_name is different? Should we extract this into a helper function so that the code doesn't get out of sync and there's a bug around different channel_names being displayed differently across our application?

Reply:
I agree with your suggestion, but there's a bug here we need to address this in a separate issue, I noticed the `.name` function didn't work on youtube channels, but it does for twitch, once that's resolved I'll refactor this using one method. (And update the `channels_controller` if necessary)

Reply:
See #1367